packages/@aws-cdk/aws-synthetics/README.md

```diff
@@ -9,8 +10,176 @@
+
+#### Cloudformation Optional Properties
+
+`successRetentionPeriod`: Specify how many days the canary should store data on successful runs in s3. It is a property in the Canary construct as well. Defaults to 31 days.
+
+`failureRetentionPeriod`: Specify how many days the canary should store data on failed runs in s3. It is a property in the Canary construct as well. Defaults to 31 days.
+
+`tags`: Specify a list of key value pairs that are associated with the canary. As is CDK convention, the Canary construct ignores this property in favor of the `Tag` class.
+
+`vpcConfig`: If the canary tests endpoints inside of a VPC, then this structure contains information on where to find that VPC. The properties of `vpcConfig` are `securityGroupIds`, `subnetIds`, and `vpcId`. The Canary construct has a property `vpc` that allows for this information to be specified similar to the Lambda construct.
+
+### Canary Code
+
+You can use three different static `Code` methods to specify the script that the canary should run. 
+
+#### From Inline 
+
+```ts
+import * as synth from '@aws-cdk/aws-synthetics';
+
+const canary = new synth.Canary(this,'my_test',{
+	handler: 'index.handler',
+	canaryName: 'mycanary',
+	code: synth.Code.fromInline('your code here'),
+});
+```
+
+#### From Asset
+
+```ts
+import * as synth from '@aws-cdk/aws-synthetics';
+
+// 'index.js' is in the directory 'canary-handler'
+const canary = new synth.Canary(this,'my_test',{
+	handler: 'index.handler',
+	canaryName: 'mycanary',
+	code: synth.Code.fromAsset('./canary-handler'),
+});
+```
+
+#### From S3 Bucket
+
+```ts
+import * as synth from '@aws-cdk/aws-synthetics';
+import * as s3 from '@aws-cdk/aws-s3';
+import * as s3Deployment from '@aws-cdk/aws-s3-deployment';
+
+// Create bucket
+const bucket = new s3.Bucket(this, 'canarybucket');
+
+// Upload assets from the canary directory to the bucket
+new s3Deployment.BucketDeployment(this, 'Deployment', {
+	sources: [s3Deployment.Source.asset('./canary')],
+	destinationBucket: bucket,
+});
+
+// 'index.js' is the file name and key in s3
+const canary = new synth.Canary(this,'my_test',{
+	handler: 'index.handler',
+	canaryName: 'mycanary',
+	code: synth.Code.fromBucket(bucket,'index.js'),
+});
+```
+
+### Canary Metrics
+
+You can add metrics to your canary through the suite of `canary.metric()` APIs. They are: 
+
+- `canary.metric(metricName, props?)`
+- `canary.metricDuration(metricName, props?)`
+- `canary.metricSucessPercent(metricName, props?)`
+- `canary.metricFailed(metricName, props?)`
+
+```ts
+import * as synth from '@aws-cdk/aws-synthetics';
+
+// 'index.js' is in the directory 'canary-handler'
+const canary = new synth.Canary(this,'my_test',{
+	handler: 'index.handler',
+	canaryName: 'mycanary',
+	code: synth.Code.fromAsset('./canary-handler'),
```

Comment:
The use of `Code.fromAsset` before you explain it will confuse the reader, keep the `inline` from the pervious example